### ISSUE TITLE
Upstream proxy

### ISSUE THREAD

**Issue by corehook** [argos83/ritm#14]
Hi!
Good job! Nice gem!

Can you explain how i can set up upstream proxy ?

net_http
```ruby
@client = Faraday.new(ssl: { verify: false }) do |conn|
        conn.adapter :net_http
        ::HTTP.Proxy('127.0.0.1', 8081)
        # TODO: support customizations (e.g. upstream proxies or different adapters)
      end
```

em-http
```ruby
@client = Faraday.new(ssl: { verify: false }, proxy: 'http://127.0.0.1:8081') do |conn|
        conn.adapter :em_http
        # TODO: support customizations (e.g. upstream proxies or different adapters)
      end
```

em-http
```ruby
@client = Faraday.new(ssl: { verify: false }, proxy: 'http://127.0.0.1:8081') do |conn|
        conn.adapter :em_http
        conn.proxy({host: 'http://127.0.0.1', port:8081})
        # TODO: support customizations (e.g. upstream proxies or different adapters)
      end
```

Faraday ignore proxy setting. Thanks!

**Comment by argos83** [comment]
Hey @corehook, that Faraday client is used only for HTTPS interception, so the reason your fix is not working might be that you are trying to intercept HTTP traffic.
Setting up an upstream proxy for the HTTP case should be done around these lines: https://github.com/argos83/ritm/blob/master/lib/ritm/proxy/proxy_server.rb#L31

I'll try to implement upstream proxies tomorrow or the day after, if you manage to fix it yourself and want to create a pull request, you are very welcomed.

**Comment by corehook** [comment]
@argos83 thx for answer!
What about universal solution for http/https upstream proxying ?
BTW my firefox does not open sites if proxy is not local ( 127.0.0.1 ) , even though I add to certificate to trusted. 
Do you tried bind not on 127.0.0.1 ? For example on other computer in network?

**Comment by argos83** [comment]
Hi @corehook, check out this branch (I still need to add test and docs before merging) but I tested manually and it works OK:  https://github.com/argos83/ritm/tree/feature/support_upstream_proxy

On that branch you should run this first to set the upstream proxy:

´´´
Ritm.conf.upstream_proxy 'http://theproxyhost:8888'
´´´
Regarding your other question, you should be able to bind ritm to 0.0.0.0 so it's reachable by other computers. And regarding Firefox, check your settings, they must be wrong, it has to work with non local proxies, however it may be skipping the proxy for the site you are targeting (try removing whatever it is in the 'No proxy for: ' section).

BTW, why do you need to set up the upstream proxy? Don't you have direct connection to the Internet?

**Comment by argos83** [comment]
This is implemented now as part or Ritm 1.0.0.

Configure an upstream proxy by doing:

```
Ritm.configure { misc[:upstream_proxy] = 'http://theupstreamproxyhost:8888' }
```
